src/lavinmq/amqp/queue/stream_queue_message_store.cr

```diff
@@ -147,6 +168,7 @@ module LavinMQ::AMQP
       private def open_new_segment(next_msg_size = 0) : MFile
         super.tap do
           drop_overflow
+          @offset_index[@segments.last_key] = @last_offset
```

Comment:
is `@last_offset` really correct here? or is it `@last_offset + 1` that will be written as the first offset to the new segment? 
```suggestion
          @offset_index[@segments.last_key] = @last_offset
```

Reply:
what about `@timestamp_index`? It isn't updated?